npm/pkg/dataplane/dataplane.go

```diff
@@ -216,6 +216,11 @@ func (dp *DataPlane) ApplyDataPlane() error {
 
 	// NOTE: ideally we won't refresh Pod Endpoints if the updatePodCache is empty
 	if dp.shouldUpdatePod() {
+		// do not refresh endpoints if the updatePodCache is empty
+		if len(dp.updatePodCache.cache) == 0 {
```

Comment:
will need to lock the cache

Reply:
updated

Reply:
Locking the updatePodCache while refreshing Endpoints would be a regression in the sense that it would block Pod controller from doing any dataplane operations (e.g. AddToSets() and applyIPSets()).

Reply:
reconfigured locks